client/apps/edit/components/content/sections/text/index.jsx

```diff
@@ -100,21 +87,52 @@ export class SectionText extends Component {
   }
 
   onChange = (editorState) => {
-    const { article, hasFeatures, onChange, section } = this.props
-    const html = convertToRichHtml(editorState, article.layout, hasFeatures)
+    const { article, hasFeatures, isContentEnd, onChange, section } = this.props
+    const convertedHtml = convertToRichHtml(editorState, article.layout, hasFeatures)
+    const html = setContentEnd(convertedHtml, isContentEnd)
 
     if (section.body !== html) {
       onChange('body', html)
     }
     this.setState({ editorState, html })
   }
 
+  blur = () => {
+    if (this.domEditor) {
+      this.domEditor.blur()
+    }
+  }
+
   focus = () => {
     if (this.domEditor) {
       this.domEditor.focus()
     }
   }
 
+  maybeResetEditor = (prevProps) => {
+    const { editing, section } = this.props
+    const { html } = this.state
+
+    const bodyHasChanged = section.body !== prevProps.section.body && section.body.length > 0
+    const bodyWasSwapped = bodyHasChanged && section.body !== html
+    const startedEditing = editing && editing !== prevProps.editing
+    const stoppedEditing = !editing && editing !== prevProps.editing
+
+    if (startedEditing || bodyWasSwapped) {
+      if (bodyHasChanged) {
+        // Re-initialize editor with new text, happens during
+        // drag/drop or changing edit section w/ handleTab or splitSection
+        this.setEditorStateFromProps()
```

Comment:
Draft works from its own state, and will not reset with new data automatically when the redux props change. This checks if props and state have gotten out of sync, for cases when sections are inserted in the middle of the list, or the order is changed by user interactions.